### PR TITLE
feat: add actionlint hook and fix current errors

### PR DESCRIPTION
**Pull request by behnazh**
This PR adds [actionlint](https://github.com/Mateusz-Grzelinski/actionlint-py) as a pre-commit hook to lint and check GitHub Actions workflows. The check runs using `make check-actionlint`. 

The new `actionlint` hook detects errors that were also addressed in https://github.com/jenstroeger/python-package-template/pull/482 and https://github.com/jenstroeger/python-package-template/pull/493, which had to be fixed here for the workflows to succeed.

### CODE REVIEW

**Comment by behnazh** [comment]
Ah, I [forgot](https://github.com/jenstroeger/python-package-template/actions/runs/4484604267/jobs/7885285239?pr=505) again that we don't create `venv` in GitHub Actions :facepalm:

**Comment by jenstroeger** [comment]
Duplicates #493, fixes #485

**Comment by jenstroeger** [comment]
What about something like [actionlint-py](https://github.com/Mateusz-Grzelinski/actionlint-py)?

**Comment by behnazh** [comment]
> What about something like [actionlint-py](https://github.com/Mateusz-Grzelinski/actionlint-py) and [shellcheck-py](https://github.com/shellcheck-py/shellcheck-py)?

Thanks. I knew about `shellcheck-py` but didn't know about `actionlint-py`. I switched to `actionlint-py` , see commit [1bf019e](https://github.com/jenstroeger/python-package-template/pull/505/commits/1bf019e79947521e5097280d601622257fb760a1).

I'm not sure if adding `shellcheck-py` makes sense in this repo though because we don't have any bash scripts :thinking:

**Comment by jenstroeger** [comment]
> I'm not sure if adding `shellcheck-py` makes sense in this repo though because we don't have any bash scripts 🤔

I think it does: `actionlint` can use `shellcheck` to check the job scripts, provided `shellcheck` is available ([docs](https://github.com/rhysd/actionlint/blob/main/docs/checks.md#check-shellcheck-integ)).

**Comment by behnazh** [comment]
> > I'm not sure if adding `shellcheck-py` makes sense in this repo though because we don't have any bash scripts thinking
> 
> I think it does: `actionlint` can use `shellcheck` to check the job scripts, provided `shellcheck` is available ([docs](https://github.com/rhysd/actionlint/blob/main/docs/checks.md#check-shellcheck-integ)).

I have already made a note on that in the `README.md` [here](https://github.com/jenstroeger/python-package-template/pull/505/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R146). Not sure if you had seen it?

File: .github/workflows/sync-with-upstream.yaml [hunk 1]
```diff
@@ -59,23 +59,23 @@ jobs:
             --exclude .git/ --exclude .github/workflows/.template_version template/ repo/
 
           # Check if the branch exists in the current repo.
-          cd repo/
-          if [ $(git rev-parse --verify origin/sync-$LATEST_VERSION 2>/dev/null) ]; then
+          cd repo/ || exit 1
+          if [ "$(git rev-parse --verify origin/sync-"""$LATEST_VERSION""" 2>/dev/null)" ]; then
             echo "Branch sync-$LATEST_VERSION already exists. Exiting..."
             exit 0
           fi
```
Comment (jenstroeger) [inline .github/workflows/sync-with-upstream.yaml:66]: Hm, so in
```bash
cd repo/ || exit 1
```
the script exits without a message and with return code `1`, and in the next line it exits with a message and return code `0`. Perhaps we can keep things a bit more consistent?

File: .github/workflows/sync-with-upstream.yaml [hunk 2]
```diff
@@ -59,23 +59,23 @@ jobs:
...
-          echo $LATEST_VERSION > .github/workflows/.template_version
+          git checkout -b sync-"$LATEST_VERSION"
+          echo "$LATEST_VERSION" > .github/workflows/.template_version
           git add .
-          git config --global user.name $USER_NAME
-          git config --global user.email $USER_EMAIL
+          git config --global user.name "$USER_NAME"
+          git config --global user.email "$USER_EMAIL"
           git config --list --global # For debug purposes.
           git commit -m "chore: sync with template"
```
Comment (jenstroeger) [inline .github/workflows/sync-with-upstream.yaml:75]: While we’re making changes here, can we change `-m` to `--message` as well, please?

**Review by jenstroeger — COMMENTED** [review]
Looks good, thanks @behnazh! Two minor comments follow…

File: .github/workflows/sync-with-upstream.yaml (hunk 2 above)
Comment (behnazh) [reply .github/workflows/sync-with-upstream.yaml]: See commit [3dd239a](https://github.com/jenstroeger/python-package-template/pull/505/commits/3dd239a6f2cec3327a94ff788e658632b6a1f5d6).

File: .github/workflows/sync-with-upstream.yaml (hunk 1 above)
Comment (behnazh) [reply .github/workflows/sync-with-upstream.yaml]: See commit [3dd239a](https://github.com/jenstroeger/python-package-template/pull/505/commits/3dd239a6f2cec3327a94ff788e658632b6a1f5d6).

**Review by jenstroeger — APPROVED** [review]

**Pull request merged**